plugins/PostProcessingPlugin/scripts/ColorMix.py

```diff
@@ -112,52 +121,53 @@ def getValue(self, line, key, default = None): #replace default getvalue due to
             return default
 
     def execute(self, data):
-        #get user variables
-        firstHeight = 0.0
-        secondHeight = 0.0
-        firstMix = 0.0
-        SecondMix = 0.0
 
         firstHeight = self.getSettingValueByKey("start_height")
         secondHeight = self.getSettingValueByKey("finish_height")
-        firstMix = self.getSettingValueByKey("mix_start_ratio")
-        SecondMix = self.getSettingValueByKey("mix_finish_ratio")
-
-        #locals
-        layer = 0
-
+        firstMix = self.getSettingValueByKey("mix_start")
+        secondMix = self.getSettingValueByKey("mix_finish")
+        modelOfInterest = self.getSettingValueByKey("objectNumber")
+        
         #get layer height
-        layerHeight = .2
+        layerHeight = 0
         for active_layer in data:
             lines = active_layer.split("\n")
             for line in lines:
                 if ";Layer height: " in line:
                     layerHeight = self.getValue(line, ";Layer height: ", layerHeight)
```

Comment:
This method doesn't work for all machines. The Layer height comment is not always added.

Moreover, CuraEngine supports different layer heights in one-at-a-time mode, but the frontend doesn't support it, so I guess that's not really a problem.